sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinExec.scala

```diff
@@ -699,4 +703,18 @@ case class StreamingSymmetricHashJoinExec(
     } else {
       Nil
     }
+
+  // This operator will evict based on the state watermark on both side of inputs; we would like
+  // to let users leverage both sides of event time column for output of join, so the watermark
+  // must be lower bound of both sides of event time column. The lower bound of event time column
+  // for each side is determined by state watermark, hence we take a minimum of (left state
+  // watermark, right state watermark, input watermark) to decide the output watermark.
+  override def produceOutputWatermark(inputWatermarkMs: Long): Long = {
```

Comment:
I am a little confused on what the input watermark for join is suppose to be.  I would expect there would be two input watermarks from both sides?

Reply:
We provide a single input watermark value via min input watermark of all upstreams. We don't try to be very tight on this - we simply use global watermark as origin watermark rather than individual watermark node as origin watermark, which is already very loose.